test_scripts/Smoke/API/037_RegisterAppInterface_PositiveCase.lua

```diff
@@ -0,0 +1,119 @@
+local requestParams = {
+	syncMsgVersion = {
+		majorVersion = 2,
+		minorVersion = 2,
+	},
+	appName = "SyncProxyTester",
+	ttsName = {
+		{
+			text ="SyncProxyTester",
+			type ="TEXT",
+		},
+	},
+	ngnMediaScreenAppName = "SPT",
+	vrSynonyms = {
+		"VRSyncProxyTester",
+	},
+	isMediaApplication = true,
+	languageDesired = "EN-US",
+	hmiDisplayLanguageDesired = "EN-US",
+	appHMIType = {
+		"DEFAULT",
+	},
+	appID = "123456",
+	deviceInfo = {
+		hardware = "hardware",
+		firmwareRev = "firmwareRev",
+		os = "os",
+		osVersion = "osVersion",
+		carrier = "carrier",
+		maxNumberRFCOMMPorts = 5
+	}
+}
+
+local function SetNotificationParams()
+	local notificationParams = {
+		application = {}
+	}
+	notificationParams.application.appName = requestParams.appName
+	notificationParams.application.ngnMediaScreenAppName = requestParams.ngnMediaScreenAppName
+	notificationParams.application.isMediaApplication = requestParams.isMediaApplication
+	notificationParams.application.hmiDisplayLanguageDesired = requestParams.hmiDisplayLanguageDesired
+	notificationParams.application.appType = requestParams.appHMIType
+	notificationParams.application.deviceInfo = {
+		name = "127.0.0.1",
+		id = config.deviceMAC,
+		transportType = "WIFI",
+		isSDLAllowed = true
+	}
+	notificationParams.application.policyAppID = requestParams.appID
+	notificationParams.ttsName = requestParams.ttsName
+	notificationParams.vrSynonyms = requestParams.vrSynonyms
+	return notificationParams
+end
+
+local allParams = {
+	requestParams = requestParams,
+	notificationParams = SetNotificationParams()
+}
+
+--[[ Local Functions ]]
+local function unregisterAppInterface(self)
+	local cid = self.mobileSession1:SendRPC("UnregisterAppInterface", { })
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { appID = self.appID, unexpectedDisconnect = false })
+	self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function RegisterAppInterface(pParams, self)
+	local CorIdRAI = self.mobileSession1:SendRPC("RegisterAppInterface", pParams.requestParams)
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered", pParams.notificationParams)
+	self.mobileSession1:ExpectResponse(CorIdRAI, { success = true, resultCode = "SUCCESS" })
+	self.mobileSession1:ExpectNotification("OnHMIStatus",
+		{ hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+	self.mobileSession1:ExpectNotification("OnPermissionsChange")
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", commonSmoke.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", commonSmoke.start)
+runner.Step("RAI, PTU", commonSmoke.registerApplicationWithPTU)
+runner.Step("Activate App", commonSmoke.activateApp)
+runner.Step("UnregisterAppInterface Positive Case", unregisterAppInterface)
+
+runner.Title("Test")
+runner.Step("RegisterAppInterface Positive Case", RegisterAppInterface, { allParams })
```

Comment:
@GetmanetsIrina Think we can omit paramter ```allParams```` here.
It's possible to use local variables and functions directly in  ```RegisterAppInterface()``` function

Reply:
@dboltovskyi , Done in 607c20fb6767f2c4c0e34e662869ef978977e810.